docs/edgeql/insert.rst

```diff
@@ -0,0 +1,335 @@
+  ...     name := "Shang-Chi",
+  ...     secret_identity := "Shaun"
+  ...   })
+  ... }
+  {default::Villain {id: d47888a0-3e7b-11ec-af13-fb68c8777851}}
+
+
+Now lets write a nested insert for a ``multi`` link.
+
+.. code-block:: edgeql-repl
+
+  db> insert Movie {
+  ...   title := "Black Widow",
+  ...   characters := {
+  ...     (select Hero filter .name = "Black Widow"),
+  ...     (insert Hero { name := "Yelena Belova"}),
+  ...     (insert Villain {
+  ...       name := "Dreykov",
+  ...       nemesis := (select Hero filter .name = "Black Widow")
+  ...     })
+  ...   }
+  ... }
+  {default::Movie {id: af706c7c-3e98-11ec-abb3-4bbf3f18a61a}}
+
+
+
+We are using :ref:`set literal syntax <ref_eql_set_constructor>` to construct a
+set literal containing several ``select`` and ``insert`` subqueries. This set
+contains a mix of ``Hero`` and ``Villain`` objects; since these are both
+subtypes of ``Person`` (the expected type of ``Movie.characters``), this is
+valid.
+
+You also can't *assign* to a computed property or link; these fields don't
+actually exist in the database;
+
+.. code-block:: edgeql-repl
+
+  db> insert Hero {
+  ...   name := "Ant-Man",
+  ...   villains := (select Villain)
+  ... };
+  error: QueryError: modification of computed link 'villains' of object type
+  'default::Hero' is prohibited
+
+.. _ref_eql_insert_with:
+
+With block
+----------
+
+In the previous query, we selected Black Widow twice: once in the
+``characters`` set and again as the ``nemesis`` of Dreykov. In circumstances
+like this, you should pull that subquery into a ``with`` block.
+
+.. code-block:: edgeql-repl
+
+  db> with black_widow := (select Hero filter .name = "Black Widow")
+  ... insert Movie {
+  ...   title := "Black Widow",
+  ...   characters := {
+  ...     black_widow,
+  ...     (insert Hero { name := "Yelena Belova"}),
+  ...     (insert Villain {
+  ...       name := "Dreykov",
+  ...       nemesis := black_widow
+  ...     })
+  ...   }
+  ... }
+  {default::Movie {id: af706c7c-3e98-11ec-abb3-4bbf3f18a61a}}
+
+
+The ``with`` block can contain an arbitrary number of clauses; later clauses
+can reference earlier ones.
+
+.. code-block:: edgeql-repl
+
+  db> with
+  ...  black_widow := (select Hero filter .name = "Black Widow"),
+  ...  yelena := (insert Hero { name := "Yelena Belova"}),
+  ...  dreykov := (insert Villain {name := "Dreykov", nemesis := black_widow})
+  ... insert Movie {
+  ...   title := "Black Widow",
+  ...   characters := { black_widow, yelena, dreykov }
+  ... }
+  {default::Movie {id: af706c7c-3e98-11ec-abb3-4bbf3f18a61a}}
+
+
+.. _ref_eql_insert_conflicts:
+
+Conflicts
+---------
+
+EdgeDB provides a general-purpose mechanism for gracefully handling possible
+exclusivity constraint violations. Consider a scenario where we are trying to
+``insert`` Eternals (the ``Movie``), but we can't remember if it already exists
+in the database.
+
+.. code-block:: edgeql-repl
+
+  db> with eternals := (
```

Comment:
You don't seem to actually be using `eternals` elsewhere in the body.  What's the intent of the query?